packages/indexer/src/hooks/useSink.ts

```diff
@@ -0,0 +1,9 @@
+import type { IndexerContext } from "../context";
+
+export function useSink<TTxnParams>({
+  context,
+}: {
+  context: IndexerContext<TTxnParams>;
+}) {
+  return context.sinkTransaction;
```

Comment:
This one should check that the `context.sinkTransaction` is not null/undefined and throw in that case. Then the value returned by it can be non nullable.